Userland/Libraries/LibGUI/EmojiInputDialog.cpp

```diff
@@ -247,13 +253,31 @@ void EmojiInputDialog::update_displayed_emoji()
                     found_match = result.score > 0;
                 }
 
-                if (found_match)
+                if (found_match) {
                     horizontal_container.add_child(*emoji.button);
+                    m_displayed_emojis.append(emoji);
+                }
             }
         }
     }
 }
 
+void EmojiInputDialog::select_first_displayed_emoji() {
+    if (m_displayed_emojis.is_empty())
+        return;
+
+    Emoji first_emoji = m_displayed_emojis.first();
+
+    StringBuilder builder;
+    Span<const u32> code_points = first_emoji.emoji.code_points;
+    for (size_t i = 0; i < code_points.size(); ++i) {
+        builder.append_code_point(code_points.at(i));
+    }
```

Comment:
Let's not rebuild the text here. Instead, you can add a `String text;` field to `EmojiInputDialog::Emoji` to store the text generated in `EmojiInputDialog::supported_emoji`. There's a `StringBuilder` in there already with this data that you can leverage.